rclpy/rclpy/client.py

```diff
@@ -69,14 +69,19 @@ def __init__(
 
         self._lock = threading.Lock()
 
-    def call(self, request: SrvTypeRequest) -> SrvTypeResponse:
+    def call(
+        self,
+        request: SrvTypeRequest,
+        timeout_sec: Optional[float] = None
+    ) -> Optional[SrvTypeResponse]:
         """
         Make a service request and wait for the result.
 
-        .. warning:: Do not call this method in a callback or a deadlock may occur.
+        .. warning:: Do not call this method in a callback, or a deadlock or timeout may occur.
```

Comment:
nit, this warning is about the danger of blocking a thread in the executor. I don't think the timeout needs to be mentioned here.

Reply:
Trivial but its better be here just in case someone have this problem and don't understand why

Reply:
@sloretz i think this is okay to add? before this timeout is added, it will block the executor thread, so eventually will be deadlock. but with timeout, i think this will be either timeout (if timeout is specified) or deadlock?